Squablr/View Controllers/LoginViewController.m

```diff
@@ -88,4 +94,50 @@ - (IBAction)didTapSignup:(id)sender {
     [self performSegueWithIdentifier:@"signupSegue" sender:nil];
 }
 
+- (IBAction)signIn:(id)sender {
+    GIDConfiguration *signInConfig;
+    signInConfig = [[GIDConfiguration alloc] initWithClientID:kClientID];
+    
+    [GIDSignIn.sharedInstance signInWithConfiguration:signInConfig
+                           presentingViewController:self
+                                           callback:^(GIDGoogleUser * _Nullable user,
+                                                      NSError * _Nullable error) {
+        if (error) {
+            return;
+        }
+        // Makes call to extract user info
+        PFUser *newUser = [PFUser user];
+        
+        // set user properties
+        newUser.username = user.profile.name;
+        newUser.email = user.profile.email;
+        newUser.password = @"password";
+        [newUser signUpInBackgroundWithBlock:^(BOOL succeeded, NSError * error) {
+            if (error != nil) {
+                if ([error.localizedDescription isEqual:@"Account already exists for this username."]) {
+                    NSString *username = user.profile.name;
+                    NSString *password = @"password";
+                    [PFUser logInWithUsernameInBackground:username password:password block:^(PFUser * user, NSError *  error) {
+                        //
+                        // Manually segue now that network call has succeeded
+                        // If sign in succeeded, display the app's main content View.
+                        [self successful];
+                    }];
+                }
+            } else {
+                // Manually segue now that network call has succeeded
+                // If sign in succeeded, display the app's main content View.
+                [self successful];
+            }
+        }];
+    }];
+}
+
+- (void) successful {
```

Comment:
Need a better function name here, this function doesn't describe what the code is doing. Should be something like "navigateToUserFeed".

Reply:
🫡